modules/govuk_ci/templates/usr/local/bin/remove_old_jenkins_jobs.sh.erb

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+dirs_to_keep='<%= (@pipeline_jobs.keys + @job_builder_jobs.map { |job| job.split("::").last }).join("\n") %>'
```

Comment:
It would be much better to put this logic in the manifest rather than in the template. The template wants to be receiving a simple list to fill in here. Putting the logic in the Puppet file will allow you to write it more clearly (e.g. by giving meaningful names to intermediate values if it helps understanding) and avoid one-liners like this.

Reply:
I completely agree with you, except that I'm not sure we can accomplish this sort of thing on our version of Puppet. We don't even have a working `keys()` method in Puppet 3 😢 

Part of the complexity lies in the two different data structures:

- `pipeline_jobs` == `{ job_name: 'foo' }`
- `job_builder_jobs` == `["foo::bar::job_name"]`

I attempted to keep things DRY by [reusing the `govuk_jenkins::job_builder::jobs` hieradata](https://github.com/alphagov/govuk-puppet/blob/acafcb6eb24b855dc5fd245ea2353b5337d55836/hieradata_aws/class/integration/ci_master.yaml#L146). But on reflection, a small bit of duplication to tidy up the template and make it readable seems a net win.

Reply:
Hmmm even the simple hash concatenation appears not to be supported, even if I [enable 'future' parser mode](https://github.com/alphagov/govuk-puppet/pull/11630/commits/6348a9671848418b12277e57b6e449e925eb4ca4). Any thoughts, @sengi?

Reply:
Oh gosh, sorry had no idea that would be a rabbithole. I wouldn't worry about it then, let's just do it in the template 😅

Reply:
Cool, I've removed those commits now. For future reference, they were https://github.com/alphagov/govuk-puppet/commit/d2093fb4477f2e001cadde4850312b13a6b6c057 and https://github.com/alphagov/govuk-puppet/commit/6348a9671848418b12277e57b6e449e925eb4ca4.

Could you re-review @sengi, and I'll look at getting this deployed today?